File: Cybersec/MyPizzaApp/backend/src/awefull_pizza_shop/webserver/schemas/pizza.py
from dataclasses import dataclass
from typing import Annotated
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field
from pydantic_core import Url

from awefull_pizza_shop.database.models.pizza import PizzaCategory


class PizzaBase(BaseModel):
    name: str
    description: str
    price: float
    image_url: Annotated[Url, Field(serialization_alias="imageUrl")]


# class PizzaCreation(PizzaBase):
#    pass


class Pizza(PizzaBase):
    model_config = ConfigDict(from_attributes=True)
    id: UUID


# Pizza class for RCE
@dataclass
class PizzaCreation:
    name: str
    description: str
    price: float
    image_url: str
    category : PizzaCategory

File: Cybersec/MyPizzaApp/frontend/AwefullPizzaShop/src/app/pizza/types/pizza.ts
import {UserComment} from './userComment';

export interface Pizza {
  id: string;              // Unique identifier for the product
  name: string;            // Name of the product
  price: number;           // Price in the appropriate currency
  imageUrl: string;        // URL of the product image
  description: string;     // Product description
}
